### PR TITLE
Export the INTERFACE_LINK_DIRECTORIES attribute for ARPACK target

### PR DESCRIPTION
**Pull request by dariomangoni**
Projects (at least in VS) returns a missing "arpack.lib"; this is because the library folder is not added in the link directories.
By adding them the issue is fixed.

### CODE REVIEW

**Comment by fghoussen** [comment]
CI down (docker KO) : we are blind at the time. No merge possible. @sylvestre : can you handle #309 ?

**Comment by wavepacket** [comment]
I just wanted to raise this issue myself.

Just one note related to this issue: The property INTERFACE_LINK_DIRECTORIES was added in CMake 3.12. The CMakeLists.txt in the root folder declares compatibility with CMake 3.0. What compatibility do you expect for your installed script?

I absolutely do not want to discourage this patch. In the worst case, you could always enclose this property in an " if (CMAKE_VERSION >= ...)"  block. It is just completely unclear to me what you expect to support.

**Comment by fghoussen** [comment]
The spirit is to be "most ahead/updated" as possible (preserving old distro ans stuffs working on old systems): rebase your PR and change minimal cmake version, if CI is OK, your patch will be merged. @sylvestre : agreed?

**Comment by wavepacket** [comment]
Another note: You provide the installed libarpack through an INTERFACE IMPORTED library.

While this works, a more semantically correct solution would be to use a SHARED IMPORTED or a STATIC IMPORTED target (depending on how arpack was built). These targets can then have a property IMPORTED_LOCATION, which points to the file that the library target describes. This would be closer to the book, I think, but may be overkill.

**Comment by fghoussen** [comment]
Sounds good to me! The INTERFACE IMPORTED was probably set by default, do not hesitate to improve it: your patch will be merged if CI runs OK! If you provide a patch, remember to impact both libarpack and lib**p**arpack. Thanks!

**Comment by wavepacket** [comment]
Actually, this is one part of the other pull request (#296) by DavidAce. I looked it over, and cannot see obvious flaws there.

**Comment by fghoussen** [comment]
Split it! Better have 1 fix for 1 PR.

**Comment by kleinhenz** [comment]
I just ran into this. Would be nice to have it just work.

File: arpack-ng-config.cmake.in [hunk 1]
```diff
@@ -21,5 +21,6 @@ set(includedir "@includedir@")
 add_library(ARPACK::ARPACK INTERFACE IMPORTED)
 set_target_properties(ARPACK::ARPACK PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "${includedir}/arpack")
```
Comment (fghoussen) [inline arpack-ng-config.cmake.in:22]: Can you add INTERFACE_INCLUDE_DIRECTORIES for PARPACK too?

File: arpack-ng-config.cmake.in [hunk 2]
```diff
@@ -21,5 +21,6 @@ set(includedir "@includedir@")
...
 set_target_properties(ARPACK::ARPACK PROPERTIES INTERFACE_LINK_LIBRARIES      "arpack")
+set_target_properties(ARPACK::ARPACK PROPERTIES INTERFACE_LINK_DIRECTORIES "${libdir}")
```
Comment (fghoussen) [inline arpack-ng-config.cmake.in:24]: Can you add INTERFACE_LINK_DIRECTORIES for PARPACK too?

**Comment by fghoussen** [comment]
Included in #350

**Pull request closed without merging**